src/memray/_memray/record_writer.h

```diff
@@ -92,6 +103,34 @@ bool inline RecordWriter::writeVarint(size_t rest)
     return writeSimpleType(next_7_bits);
 }
 
+bool inline RecordWriter::writeSignedVarint(ssize_t val)
+{
+    // protobuf style "zig-zag" encoding
+    size_t zigzag_val = (static_cast<size_t>(val) << 1)
+                        ^ static_cast<size_t>(val >> std::numeric_limits<ssize_t>::digits);
+    return writeVarint(zigzag_val);
+}
+
+template<typename T>
+bool inline RecordWriter::writeIntegralDelta(T* prev, T new_val)
+{
+    ssize_t delta = new_val - *prev;
+    *prev = new_val;
+    return writeSignedVarint(delta);
+}
+
+bool inline RecordWriter::writeStringDelta(std::string* prev, const char* new_str)
```

Comment:
I wonder if this makes a lot of sense. We don't emit segment headers and frame indexes that often and the complexity of writing these is quite big compared with varints. Did you capture by any chance stats on how efficient this is (how long the suffixes tend to be)?

Reply:
I skipped it for function names based on the low utility, but it works pretty well for file names. The prefixes tend to be reasonably long - often we see long runs of `/usr/lib/python3.10/site-packages/...` where this encoding lets us avoid sending that prefix (the `site-packages` directory) over and over.

_BUT_, you're right that we don't send strings very often, so even though this does a pretty good job of compressing the strings, it makes little difference to the overall size of the file. And you're right that it more complex than the varints - I made more mistakes while writing it, and it took longer to debug.

For my test file generated with:
```
memray run --native -m test test_list test_ast
```
the total size of all filenames in the capture file (without any delta compression) is ~200kb, and the total file is about 80mb, so even if we were able to elide them entirely, it would save at most 0.2% of the total file size.

Alright, that's enough to convince me - I'm gonna drop the delta encoding of strings. I'll keep it in a branch should we decide to resurrect it, but for now, even though it works well for file names, we don't write file names often enough to justify the complexity.